### DISCUSSION TITLE
Obtaining listing variations with API v3

### DISCUSSION THREAD

**Discussion by alejandroLoz** [etsy/open-api#948] [Q&A] [answered]
Heyall!

I've been trying to obtain variations via API without much luck.
I've created a product with variations on Etsy's admin UI as follows:

![image](https://user-images.githubusercontent.com/13033000/230686745-42b69f6e-34e3-4436-abe8-6216df02f943.png)

But when I call [getListingsByShop](https://developer.etsy.com/documentation/reference#operation/getListingsByShop) the only trace of variations I get is the "has_variations" field but I can't bet the actual variations. 
I checked the docs and the only way to obtain the variations is using the [getShopReceiptTransaction](https://developer.etsy.com/documentation/reference#operation/getShopReceiptTransaction) endpoint which is not something that I can do as I want to obtain the variations of a listing before anyone places an order.

Is there any way to do this?

Thanks!!!

**Comment by Gareth064** [comment, marked as answer]
You need to Include Inventory and then dig into the Product data inside the inventory to find the Products , and then Properties of those products are what you are probably looking for.

I would say, and others can correct me if I'm wrong, there is a shift in terminology whereby a "Variation" is actually a "Product"

![image](https://user-images.githubusercontent.com/22307223/230689986-ee0ebefd-fff8-44af-ad0a-dcdf0e93f831.png)
![image](https://user-images.githubusercontent.com/22307223/230690180-9f25d714-df96-46e0-ada3-4992ba29ee93.png)

**Reply by showtroylove** [reply]
@alejandroLoz based on the image you provided, you want the **[getListingInventory](https://developers.etsy.com/documentation/reference/#operation/getListingInventory)** endpoint.  It will return all the **variations** of your listing.  You may also want to view / update the variation images.  The latter tells Etsy how to display the appropriate image associated with that variation of your listing (i.e. [getListingVariationImages](https://developers.etsy.com/documentation/reference/#tag/ShopListing-VariationImage)).

**Comment by Grumpier** [comment]
Does anyone know why getListingsByShop returns all of the property values EXCEPT the variation image ids? It's a real pain to call getListingVariationImages for each listing and then join back on property_id.
